backend/src/main/java/com/bakdata/conquery/models/auth/AuthorizationHelper.java

```diff
@@ -146,17 +161,18 @@ public static void removePermission(PermissionOwner<?> owner, Permission permiss
 		}
 		return permissions;
 	}
-
+	
 	/**
-	 * Returns the groups a user belongs to.
+	 * Returns a list of the effective permissions. These are the permissions of the owner and
+	 * the permission of the roles it inherits. The query can be filtered by the Permission domain.
+	 * @return Owned and inherited permissions.
 	 */
-	public static Collection<Group> getGroupsOf(User user, MasterMetaStorage storage) {
-		List<Group> userGroups = new ArrayList<>();
-		for (Group group : storage.getAllGroups()) {
-			if (group.containsMember(user)) {
-				userGroups.add(group);
-			}
-		}
-		return userGroups;
+	public static Set<WildcardPermission> getEffectiveUserPermissions(UserId userId, List<String> domainSpecifier, MasterMetaStorage storage) {
+		Set<ConqueryPermission> permissions = getEffectiveUserPermissions(userId, storage);
+		return permissions.stream()
```

Comment:
wenn die Häufig verwendet wird, solltest du mit dem Gedanken spielen das als for-Schleife zu implementieren

Reply:
sollte nur einmal aufgerufen werden, wenn das frontend startet